graph-builder/src/main.rs

```diff
@@ -35,6 +39,9 @@ fn main() -> Result<(), Error> {
     let registry: prometheus::Registry =
         metrics::new_registry(Some(config::METRICS_PREFIX.to_string()))?;
 
+    // Enable tracing
+    init_tracer("graph-builder")?;
```

Comment:
Should this be configurable?

Reply:
Its done once per binary, I think its safe to hardcode the name here

Reply:
I meant the initialization of the tracer in general. Related to this is this question: how is the endpoint configured to which the traces are sent?

Reply:
Current implementation is for developers only, thus hardcoded endpoint in the init_tracer. I'm planning to update this later on, once Jaeger gets deployed on app-sre side and we sort out the connection details.

Not sure if I should expose configuration early - it might have to be reworked later anyway

Reply:
What happens if the endpoint where traces are sent isn't available? Or is it pull-based like Prometheus?

Reply:
Its fire-and-forget. If the collector doesn't respond then no error thrown and its completely invisible to the app.

Not sure if I like that when its deployed on prod, but probably it can be tweaked later

Reply:
Please let's take the time and tweak it before we land it. Is there a way it can verify that the controller is reachable?
If so, I think that `init_tracer` should throw an error if it's not.

Either way we should only call `init_tracer` if tracing has been configured. I think it makes sense to configure the controller socket as well. WDYT?

Reply:
> Either way we should only call init_tracer if tracing has been configured

So you're proposing a flag to enable tracing and require Jaeger? 
That doesn't align well with the idea of open tracing - you should be able to deploy Jaeger and get trace data without Cincinnati reconfiguration (and pod redeploy).

I don't mind exposing a better config option for tracing later, once our production would have Jaeger deployed and default setting won't suit us (or customers).

Reply:
>  That doesn't align well with the idea of open tracing - you should be able to deploy Jaeger and get trace data without Cincinnati reconfiguration (and pod redeploy).

It does come with some unnecessary load of sending the packets and having the kernel discard them since there's no one listening. I wasn't aware that it's meant to be a plug & play system and that it's designed this way. I'm not really fond of this design, but I can accept it.